test_runner/src/main/kotlin/ftl/args/ArgsHelper.kt

```diff
@@ -150,4 +172,24 @@ object ArgsHelper {
         // Allow users control over projectId by checking using Google's logic first before falling back to JSON.
         return ServiceOptions.getDefaultProjectId() ?: serviceAccountProjectId()
     }
+
+    // https://stackoverflow.com/a/2821201/2450315
+    private val envRegex = Pattern.compile("\\$([a-zA-Z_]+[a-zA-Z0-9_]*)")
+    private fun substituteEnvVars(text: String): String {
```

Comment:
we should check this works for duplicate environment variables via more JUnit tests. There are probably other edge cases as well. For example the `${HOME}` syntax which we should probably document as unsupported.

> $HOME/$HOME/tmp

Reply:
We should probably _WARN_ but not fail for duplicate env variables. The warning is to notify the user that he's duplicated an env variable. Not failing, in case its intentional.

Regardless, I don't think we should handle custom env vars. That would likely be a pain to maintain.

Reply:
bash doesn't warn on duplicates. I'm using bash as the model for the parsing logic.